gef.py

```diff
@@ -1059,6 +1060,8 @@ def set_gef_setting(name, value, _type=None, _desc=None):
     are not provided."""
     global __config__
 
+    reset_all_caches()
```

Comment:
That's way too violent. If you want to cache the setting, it's best to use a different cache.

TBH I don't see the point in caching like this, at most `get_gef_setting(<setting>)` should be called at most once when it's being used like that's pretty harmless in terms of speed, and flushing the cache is not.

Reply:
The cache is flushed every step/stop anyway.

get_gef_setting tends to be called a lot for colours -- This was my attempt to make that quicker.

#350 would fix this, but first let me see how many times the settings are queried to see if it's worth it. If you look at the times above, the change that adds this + lru_cache speeds up the context

Reply:
You're totally right about this: Where I meant to put this is in `ConfigCommand`: This way it's only cleared when a user changes a setting, not when gef does internally. My goal was for the caching above and also so that it still worked with e.g. changing the theme and doing `ctx` to see the result. What do you think about putting it there?